diff -ruN gamescope-3.16.28.orig/src/wlserver.cpp gamescope-3.16.28/src/wlserver.cpp
--- gamescope-3.16.28.orig/src/wlserver.cpp	2026-08-28 00:03:04.000000000 +0200
+++ gamescope-3.16.28/src/wlserver.cpp	2026-09-17 15:34:31.971400122 +0200
@@ -41,6 +41,7 @@
 #include <wlr/util/log.h>
 #include <wlr/xwayland/server.h>
 #include <wlr/types/wlr_xdg_shell.h>
+#include <wlr/types/wlr_idle_notify_v1.h>
 #include <wlr/types/wlr_relative_pointer_v1.h>
 #include <wlr/types/wlr_pointer_constraints_v1.h>
 #include <wlr/types/wlr_layer_shell_v1.h>
@@ -283,6 +284,7 @@
 
 static void bump_input_counter()
 {
+	wlr_idle_notifier_v1_notify_activity(wlserver.wlr.idle_notifier, wlserver.wlr.seat);
 	inputCounter++;
 	nudge_steamcompmgr();
 }
@@ -2071,6 +2073,8 @@
 
 	wl_signal_add( &wlserver.wlr.compositor->events.new_surface, &new_surface_listener );
 
+	wlserver.wlr.idle_notifier = wlr_idle_notifier_v1_create(wlserver.display);
+
 	create_ime_manager( &wlserver );
 
 	create_reshade();
diff -ruN gamescope-3.16.28.orig/src/wlserver.hpp gamescope-3.16.28/src/wlserver.hpp
--- gamescope-3.16.28.orig/src/wlserver.hpp	2026-08-28 00:03:04.000000000 +0200
+++ gamescope-3.16.28/src/wlserver.hpp	2026-09-17 15:35:17.304992096 +0200
@@ -131,6 +131,7 @@
 		struct wl_listener device_change_listener = {};
 
 		std::vector<std::unique_ptr<gamescope_xwayland_server_t>> xwayland_servers;
+		struct wlr_idle_notifier_v1 *idle_notifier;
 	} wlr;
 	
 	struct wlr_surface *mouse_focus_surface;
